### PR TITLE
 feat(react-chart): add ability to specify new types of axes

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: packages/dx-react-chart/docs/reference/scale.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { Scale } from '@devexpress/dx-react-chart';
+```
+
+## User Reference
+
+### Properties
+
+Name | Type | Default | Description
+-----|------|---------|------------
+extension? | Array&lt;[scaleOptions](#scaleoptions)&gt; | [{type: 'linear', constructor: () => object}, {type: 'band', constructor: () => object}] | A array of scales with constructors.
```
Comment (kvet) [inline packages/dx-react-chart/docs/reference/scale.md]: Why not *extensions*

File: packages/dx-chart-core/src/utils/scale.test.js [hunk 2]
```diff
@@ -29,34 +25,28 @@ describe('Create scale', () => {
     jest.clearAllMocks();
   });
 
-  it('should be created from d3 scale linear', () => {
-    const scale = createScale(domainOptions, width, height);
+  it('should creat scale', () => {
```
Comment (alexey-semikozov) [inline packages/dx-chart-core/src/utils/scale.test.js]: *create

File: packages/dx-react-chart-demos/src/demo-sources/chart-basic/stacked-area.jsxt [hunk 3]
```diff
@@ -55,6 +57,7 @@ export default class Demo extends React.PureComponent {
             stack="a"
           />
           <Stack />
+          <Scale extension={[{ type: 'band', constructor: () => scalePoint() }]} />
```
Comment (alexey-semikozov) [inline packages/dx-react-chart-demos/src/demo-sources/chart-basic/stacked-area.jsxt]: We can rewrite:
```
() => scalePoint()
```
 to 
```
scalePoint
```

File: packages/dx-react-chart-demos/src/demo-sources/chart-basic/streamgraph.jsxt [hunk 4]
```diff
@@ -60,7 +60,7 @@ export default class Demo extends React.PureComponent {
           data={chartData}
           style={{ paddingLeft: '20px' }}
         >
-          <ArgumentAxis name="argument" />
+          <ArgumentAxis name="argument" tickFormat={() => tick => tick} />
```
Comment (alexey-semikozov) [inline packages/dx-react-chart-demos/src/demo-sources/chart-basic/streamgraph.jsxt]: Why are we use this property?

File: packages/dx-react-chart/src/plugins/scale.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { Plugin, Getter } from '@devexpress/dx-react-core';
+import { domains, computedExtension } from '@devexpress/dx-chart-core';
+
+// eslint-disable-next-line react/prefer-stateless-function
```
Comment (alexey-semikozov) [inline packages/dx-react-chart/src/plugins/scale.jsx]: You are sure about this line

File: packages/dx-react-chart/docs/guides/plugin-overview.md [hunk 6]
```diff
@@ -21,11 +26,18 @@ The plugins below render extra elements like a grid, legend, axes, etc.
 - `Legend` - renders a legend
 - `Title` - renders a title
 
+## Data Processing Plugins
+
+- `Scale` - process data for futher rendering axes and series.
```
[the remark this answers is not in the document]
Comment (Krijovnick) [reply packages/dx-react-chart/docs/guides/plugin-overview.md]: Yes, we can

File: packages/dx-react-chart/docs/guides/axes.md [hunk 7]
```diff
@@ -8,18 +8,25 @@ The following plugins implement the axes and grid:
...
+- `Scale` - processes data
 
 ## Basic Setup
 
 Import the plugins listed above to set up a Chart with two axes and a grid.
 
 ## Axis Types
 
-The default axis type is linear which is sutable for numeric data. The following example illustrates this axis type:
+The default axis type is linear. It is sutable for numeric data. The following example shows the Grid with linear axis:
```
Comment (dirk-pieterse) [inline packages/dx-react-chart/docs/guides/axes.md]: sutable?

File: packages/dx-react-chart/docs/guides/axes.md [hunk 8]
```diff
@@ -8,18 +8,25 @@ The following plugins implement the axes and grid:
...
 .embedded-demo({ "path": "chart-basic/default-axis", "showThemeSelector": true })
 
-The band axis type is used to display string values. To apply it, set the axis' `type` property to `"band"` as demonstrated in the following example:
+The band axis type is used to display string values. Set the axis' `type` property to `"band"` as demonstrated in the following example to use a band axis:
 
 .embedded-demo({ "path": "chart-basic/band-axis", "showThemeSelector": true })
+
+## Custom Axis Types
+
+You can also use custom axis types. For this, assign an array of scale options objects to the `Scale` plugin's `extensions` property. The following example demonstrates how to use a the logarithmic scale with d3 plugins:
```
Comment (dirk-pieterse) [inline packages/dx-react-chart/docs/guides/axes.md]: "use a the logarithmic scale" a the?

Comment (dirk-pieterse) [reply packages/dx-react-chart/docs/guides/axes.md]: "scale options objects"?

File: packages/dx-react-chart/docs/guides/plugin-overview.md [hunk 9]
```diff
@@ -1,8 +1,16 @@
...
+
+- **UI plugins**  
+ Render UI elements using the provided data.
+ 
+- **Data processing plugins**  
+ Transform data passed to the Chart component.
+
+## UI Plugins
 
 The following plugins render a corresponding series:
```
Comment (dirk-pieterse) [inline packages/dx-react-chart/docs/guides/plugin-overview.md]: render a corresponding series?

File: packages/dx-react-chart/docs/guides/plugin-overview.md [hunk 10]
```diff
@@ -21,11 +29,19 @@ The plugins below render extra elements like a grid, legend, axes, etc.
 - `Legend` - renders a legend
 - `Title` - renders a title
 
+## Data Processing Plugins
+
+- `Scale` - extends user data with service information that is required to render axes and series
+- `Stack` - transforms user data to display display series points side-by-side or one under another other
```
Comment (dirk-pieterse) [inline packages/dx-react-chart/docs/guides/plugin-overview.md]: display display?

File: packages/dx-react-chart/docs/guides/plugin-overview.md [hunk 11]
```diff
@@ -13,19 +21,27 @@ The following plugins render a corresponding series:
...
 
-The plugin order is important because plugins are rendered in the same order they are defined in the Chart component. For example, if the `BarSeries` plugin precedes the `LineSeries`, the line series overlays the bar series.
+## Plugin Order
+
+The plugin order is important. UI plugins are rendered in the same order as they are defined in the Chart component. For example, if the `BarSeries` plugin precedes the `LineSeries`, the line series overlays the bar series.
 
 .embedded-demo({ "path": "chart-basic/combination-series", "showThemeSelector": true })
 
+Note that the data processing plugin `Scale` should be defined after series and axes plugins because it requires options defined in these plugins for correct work.
```
Comment (dirk-pieterse) [inline packages/dx-react-chart/docs/guides/plugin-overview.md]: for correct work?

File: packages/dx-react-chart/docs/guides/plugin-overview.md [hunk 12]
```diff
@@ -13,19 +21,27 @@ The following plugins render a corresponding series:
...
+
 The Chart's plugins use special components to render the UI. You can implement your component suite or use a predefined one:
```
Comment (dirk-pieterse) [inline packages/dx-react-chart/docs/guides/plugin-overview.md]: use a predefined one -> use a predefined **suite**:

File: packages/dx-react-chart/docs/reference/argument-axis.md [hunk 13]
```diff
@@ -27,7 +27,8 @@ tickSize? | number | 5 | The tick size.
 position? | 'bottom' &#124; 'top' | 'bottom' | The axis position.
 name? | string | | The axis name.
 indentFromAxis? | number | 10 | The indent from the axis.
-type? | 'band' &#124; 'linear' | | Axis type.
+type? | 'band' &#124; 'linear' | 'linear' | Axis type.
+tickForamt? | (scale: object) => (tick: string) => string  | | A function that returns a tick formatter function.
 rootComponent | ComponentType&lt;[ArgumentAxis.RootProps](#argumentaxisrootprops)&gt; | |  A component that renders the axis root layout.
 tickComponent | ComponentType&lt;[ArgumentAxis.TickProps](#argumentaxistickprops)&gt; | | A component that renders a tick.
 labelComponent | ComponentType&lt;[ArgumentAxis.LabelProps](#argumentaxislabelprops)&gt; | | A component that renders the axis label.
```
Comment (dirk-pieterse) [inline packages/dx-react-chart/docs/reference/argument-axis.md]: "that renders the axis label" Should this be 'that renders the **axis's** label'?

**Pull request merged**